File: Java/Dependency Inversion Principle/Haber.java
import java.util.ArrayList;
import java.util.List;

public class Haber {
    private List<İletisim> iletisimListesi;
    private Mail mail;
    private Bildirim bildirim;

    public Haber() {
        iletisimListesi = new ArrayList<>();
        mail = new Mail();
        bildirim = new Bildirim();
        iletisimListesi.add(mail);
        iletisimListesi.add(bildirim);
    }

    public void bilgilendir(Kullanici kullanici, String mesaj) {
        for (İletisim iletisim : iletisimListesi) {
            iletisim.bilgiGonder(kullanici, mesaj);
        }
    }
}

File: Python/Dependency Inversion Principle/Haber.py
from Mail import Mail
from Bildirim import Bildirim

class Haber:
    
    def __init__(self):
        self.iletisimListesi=[]
        self.mail=Mail()
        self.bildirim=Bildirim()
        self.iletisimListesi.append(self.mail)
        self.iletisimListesi.append(self.bildirim)
        
    def Bilgilendir(self,kullanici,mesaj):
        
        for iletisim in self.iletisimListesi:
            iletisim.BilgiGönder(kullanici,mesaj)
    
    
    